### DISCUSSION TITLE
case_insensitive search does not search for upper case keywords

### DISCUSSION THREAD

**Discussion by uchajk** [yajra/laravel-datatables#2899] [Q&A]
`Case insensitive will search the keyword in lower case format.
SQL: LOWER(column) LIKE LOWER(keyword)`
`'case_insensitive' => true,`

I've got an issue with this part of the documentation. Particular column in my table is in upper case and so when searching in lower case: 
`$keyword = 'abc'`
`$column = LOWER('ABC')`
this works just fine

When searching upper case though:
`$keyword = 'ABC'`
`$column = LOWER('ABC')`
does not match the column. LOWER(keyword) is not applied.

**Comment by yajra** [comment]
@uchajk can you provide snippets to reproduce the issue? Are you using [filterColumn](https://yajrabox.com/docs/laravel-datatables/10.0/filter-column#main-content)?

**Reply by challgren** [reply]
This issue occurs when `starts_with` is enabled. The function `prepareKeyword` of `QueryDataTable` returns early due to `$this->config->isStartsWithSearch()` returning true and not executing `$this->config->isCaseInsensitive()` check first.

**Reply by yajra** [reply]
PR https://github.com/yajra/laravel-datatables/pull/2916 merged and released, thanks @challgren!

**Reply by challgren** [reply]
I would almost propose giving the developer the option to use Upper or Lower on either side of the equation. In our instance our database is all uppercase, and we really don't need a `UPPER(column_name)` or `LOWER(column_name)` which if I recall can be expensive because the index is lost on a function column, unless a functional index is added by the developer.

**Reply by yajra** [reply]
I agree, I was also thinking the same lately on how I can approach the indexing issue given the nature of wildcard search. Just learned it recently from Laracasts lol >.<
